### ISSUE TITLE
Stacks token issue, impossible to send with Ledger connected wallet

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/blockstack-app#5025]
https://explorer.hiro.so/address/SP1BZBQCMWNF7Q6F5585Y3R6Z9FXSN5SY9J03VGCW.FrodoSaylorKeanuPepe10Inu-token?chain=mainnet

The user tries to send this token from their extension version 6.28.0 connected with Ledger nano X (latest firmware)
Click send
Select token
Input amount (with our without decimals, small or big amount)
Adds Stacks address
Confirm and see
![image](https://github.com/leather-wallet/extension/assets/33360391/e4577147-f1a3-47bd-9fa3-1180aa74ffd4)

With 500 amount

![image](https://github.com/leather-wallet/extension/assets/33360391/2d35e04f-bc09-4809-ac5e-65030f52cea6)

![image](https://github.com/leather-wallet/extension/assets/33360391/7673525b-001d-4d7d-b526-fcf98d17170d)

The user is now trying to deploy the token to a software wallet to see if there is a difference.

**Comment by markmhendrickson** [comment]
This seems like a P2 since the user is blocked?

**Comment by 314159265359879** [comment]
Yes, you are right a P2. The user reported resolving the issue by deploying another contract and giving the tokens 6 decimals instead of 18. This time they only tested it with a software wallet.

I think this is a related issue https://github.com/leather-wallet/extension/issues/4761 (token with two decimals) but that was not completely stuck can only be send in small amounts.

**Comment by pete-watters** [comment]
@314159265359879 : Do you know how I can get some of this token? I checked and it's not an ALEX token I can swap to. 

If you have some could you let me know how to get some or even if you could send some to the development wallet that will help me more easily reproduce and fix this

**Comment by 314159265359879** [comment]
I don't think the token was ever distributed due to the issue. I have created a copy of the contract, changed the receiving address of the token to our dev wallet. You can use those tokens to reproduce. 

Contract 1 (doc 1) is what I deployed (currently pending)
```
;; ---------------------------------------------------------
;; FrodoSaylorKeanuPepe10Inu Token for testing
;; ---------------------------------------------------------

(impl-trait 'SP3FBR2AGK5H9QBDH3EEN6DF8EK8JY7RX8QJ5SVTE.sip-010-trait-ft-standard.sip-010-trait)

(define-fungible-token FrodoSaylorKeanuPepe10Inu)
(define-constant contract-owner tx-sender)

(define-data-var token-uri (optional (string-utf8 256)) none)

(define-constant ERR_GIT_GOOD (err u100))

(define-public (transfer (amount uint) (sender principal) (recipient principal) (memo (optional (buff 34))))
  (begin
    (asserts! (is-eq tx-sender sender) ERR_GIT_GOOD)
    (try! (ft-transfer? FrodoSaylorKeanuPepe10Inu amount sender recipient))
    (match memo to-print (print to-print) 0x)
    (ok true)
  )
)

(define-read-only (get-balance (owner principal))
  (ok (ft-get-balance FrodoSaylorKeanuPepe10Inu owner))
)

(define-read-only (get-name)
  (ok "FrodoSaylorKeanuPepe10Inu")
)

(define-read-only (get-symbol)
  (ok "ETHEREUM")
)

(define-read-only (get-decimals)
  (ok u18)
)

(define-read-only (get-total-supply)
  (ok (ft-get-supply FrodoSaylorKeanuPepe10Inu))
)

(define-read-only (get-token-uri)
    (ok (var-get token-uri)
    )
)

(define-public (set-token-uri (value (string-utf8 256)))
  (if (is-eq tx-sender contract-owner)
    (ok (var-set token-uri (some value)))
    (err ERR_GIT_GOOD)
  )
)

(define-public (send-many (recipients (list 1000 { to: principal, amount: uint, memo: (optional (buff 34)) })))
  (fold check-err (map send-token recipients) (ok true))
)

(define-private (check-err (result (response bool uint)) (prior (response bool uint)))
  (match prior ok-value result err-value (err err-value))
)

(define-private (send-token (recipient { to: principal, amount: uint, memo: (optional (buff 34)) }))
  (send-token-with-memo (get amount recipient) (get to recipient) (get memo recipient))
)

(define-private (send-token-with-memo (amount uint) (to principal) (memo (optional (buff 34))))
  (let ((transferOk (try! (transfer amount tx-sender to memo))))
    (ok transferOk)
  )
)

(begin
  (try! (ft-mint? FrodoSaylorKeanuPepe10Inu u120138879310000000000000000 'SP2417H88DQFN7FNDMSKM9N0B3Q6GNGEM40W7ZAZW))
)
```

The owner deployed a second contract that fixed the issue (document 2)
https://explorer.hiro.so/txid/SPPK49DG7WR1J5D50GZ4W7DYYWM5MAXSX0ZA9VEJ.FrodoSaylorKeanuPepe10Inu-token-v69?chain=mainnet

They thought the number of decimals may have been the issue. That is a notable change in the second contract (6 decimals instead of the original 18 decimals). Another change was the token-uri. None is available in the initial contract code for contract 1 but for contract 2 there is already a link in the initial deployment code. Uri can always be updated by the contract owner. We've seen issues with metadata not getting picked up after an update by the token metadata indexer. Although I am not sure if that could have caused this. The final difference is the total minting amount it was changed with the new decimal regime. Both values are well within the maximums in Clarity for uints.
